test/battle/move_effect/hit_set_remove_terrain.c

```diff
@@ -0,0 +1,83 @@
+#include "global.h"
+#include "test/battle.h"
+
+ASSUMPTIONS
+{
+    ASSUME(gBattleMoves[MOVE_ELECTRIC_TERRAIN].effect == MOVE_ELECTRIC_TERRAIN);
+    ASSUME(gBattleMoves[MOVE_PSYCHIC_TERRAIN].effect == EFFECT_PSYCHIC_TERRAIN);
+    ASSUME(gBattleMoves[MOVE_GRASSY_TERRAIN].effect == EFFECT_GRASSY_TERRAIN);
+    ASSUME(gBattleMoves[MOVE_MISTY_TERRAIN].effect == EFFECT_MISTY_TERRAIN);
+    ASSUME(gBattleMoves[MOVE_STEEL_ROLLER].effect == EFFECT_RECOIL_IF_MISS);
+    ASSUME(gBattleMoves[MOVE_ICE_SPINNER].effect == EFFECT_RECOIL_IF_MISS);
```

Comment:
Neither of them are `EFFECT_HIT_SET_REMOVE_TERRAIN` ._.

Reply:
my bad, sorry
```
    ASSUME(gBattleMoves[MOVE_STEEL_ROLLER].effect == EFFECT_HIT_SET_REMOVE_TERRAIN);
    ASSUME(gBattleMoves[MOVE_ICE_SPINNER].effect == EFFECT_HIT_SET_REMOVE_TERRAIN);
```

Reply:
I know, it's an issue with `ASSUMPTIONS`: #3043

Reply:
so basically assumptions currently don't matter?

Reply:
They should still be fixed